### PR TITLE
TINY-6294: Fixed TrimNode.trimNode() removing nodes it shouldn't

### LINKED ISSUE #4995: Tiny removes spaces inside the editor content when changing certain formats
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Tiny removes spaces inside the editor content when changing certain formats.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**
Using the full featured demo: https://www.tiny.cloud/docs/demo/full-featured/
1. Write two words with a space in between.
2. Select first word, apply bold format
3. Select second word, apply underline, italic and bold
4. While the second word is still selected, unapply underline, italic and bold
-> space in between the two words is gone
evidence

**What is the expected behavior?**
Tiny must not alter the content of the editor.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
I am using Google Chrome Version 74.0.3729.131 (Official Build) (64-bit) on Arch Linux 5.0.13.
I did not test it in other versions.

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6294

Description of Changes:

When investigating TINY-6251 (https://github.com/tinymce/tinymce/pull/5937) we found a number of bugs with the with the `TrimNode` module. We deemed they were probably too risky to include in 5.4.2, so we're including a partial fix there and this should be a more in-depth fix. The idea behind this is to re-use an existing isEmpty check instead of trying to reimplement it.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):
Fixes #4995

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/dom/TrimNode.ts [hunk 1]
```diff
@@ -5,20 +5,30 @@
...
-import * as ElementType from './ElementType';
+import DOMUtils from '../api/dom/DOMUtils';
 import * as NodeType from './NodeType';
+import * as Empty from './Empty';
 
-const surroundedBySpans = function (node) {
-  const previousIsSpan = node.previousSibling && node.previousSibling.nodeName === 'SPAN';
-  const nextIsSpan = node.nextSibling && node.nextSibling.nodeName === 'SPAN';
+const isSpan = (node: Node): node is HTMLSpanElement =>
+  node && node.nodeName.toLowerCase() === 'span';
```
Comment (ashfordneil) [inline modules/tinymce/src/core/main/ts/dom/TrimNode.ts:16]: Are we expecting people to pass null or undefined in here? I think we should either remove the `node &&` or be explicit with the types that we're expecting `node?: Node` or `node: Node | null`

Comment (ashfordneil) [reply modules/tinymce/src/core/main/ts/dom/TrimNode.ts:16]: Update, given we want to use it with previousSibling, it should be `node: Node | null`... not sure how it's compiling now, that's a worry

Comment (lnewson) [reply modules/tinymce/src/core/main/ts/dom/TrimNode.ts:16]: Take a look at the latest changes... It's no longer used for that, however you are right we don't need the `node &&` check anymore.

Comment (ashfordneil) [reply modules/tinymce/src/core/main/ts/dom/TrimNode.ts:16]: I saw the new commit just as I was submitting the review. Latest version looks fine.

File: modules/tinymce/src/core/main/ts/dom/Empty.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const isWhitespace = function (rootNode: Node, node: Node) {
 };
 
 const isNamedAnchor = function (node: Node) {
-  return NodeType.isElement(node) && node.nodeName === 'A' && node.hasAttribute('name');
+  return NodeType.isElement(node) && node.nodeName === 'A' && !node.hasAttribute('href') && (node.hasAttribute('name') || node.hasAttribute('id'));
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/dom/Empty.ts:25]: Yes, it should be if it doesn't have a `href` attribute and does have either a name or id attribute.

**Pull request merged**